templates/connect-inject-deployment.yaml

```diff
@@ -94,6 +94,13 @@ spec:
                 -health-checks-reconcile-period={{ .Values.connectInject.healthChecks.reconcilePeriod }} \
                 {{- end }}
                 -cleanup-controller-reconcile-period={{ .Values.connectInject.cleanupController.reconcilePeriod }} \
+                {{- if (or .Values.connectInject.metrics.defaultEnabled (and .Values.global.metrics.enabled (eq (.Values.connectInject.metrics.defaultEnabled | toString) "-"))) }}
```

Comment:
I think there is still value in having defaults for port, etc, even if by default metrics are **disabled**.

e.g. what if I want metrics to be disabled by default so I can enable them for specific pods via annotation but for all of those pods, I want the default scraping port to be 9102.

In that case I'd like to be able to set `defaultScrapePort: 9102`, `defaultEnabled: false`

Reply:
That does sound like a reasonable use case, and great catch! 

The rest of the metrics values (`default-enable-merging`, etc) say "Only applicable if metrics.defaultEnabled is true". I think this use case makes sense, so I can remove that sentence from the other metrics values, and that means those flags with those default values will always be passed to the connect-inject command, independent of the `defaultEnabled` value, which seems fine.

Reply:
Sounds good!